### PR TITLE
feat(performance): Expose transaction thresholds in relay config [INGEST-999] [INGEST-589] 

### PR DESCRIPTION
**Pull request by jjbayer**
In order to track user misery / apdex as a metric, we need to set a metrics tag on the user metric when extracting the metric from the transaction payload in Relay.
For this, the appropriate project config needs to be exposed.

### CODE REVIEW

File: src/sentry/relay/config.py [hunk 1]
```diff
@@ -397,6 +400,47 @@ def _filter_option_to_config_setting(flt, setting):
...
 
 
+class _TransactionThreshold(TypedDict):
+    metric: Optional[str]  # Either 'duration' or 'lcp'
+    threshold: float
+
+
+class _TransactionThresholdConfig(TypedDict):
+    projectThreshold: _TransactionThreshold
+    transactionThresholds: Mapping[str, _TransactionThreshold]
```
Comment (untitaker) [inline src/sentry/relay/config.py:410]: what's this mapping? What are the keys?

File: src/sentry/relay/config.py [hunk 2]
```diff
@@ -397,6 +400,47 @@ def _filter_option_to_config_setting(flt, setting):
...
+    return {
+        "projectThreshold": project_threshold,
+        "transactionThresholds": {
+            threshold.transaction: {
+                "metric": TRANSACTION_THRESHOLD_KEYS[threshold.metric],
+                "threshold": threshold.threshold,
+            }
+            for threshold in project.projecttransactionthresholdoverride_set.all()
+        },
+    }
```
Comment (jjbayer) [inline src/sentry/relay/config.py:441]: @shruthilayaj are `ProjectTransactionThreshold` and `ProjectTransactionThresholdOverride` the only two sources where transaction thresholds can be defined, or do I still need to check the "sentry:apdex_threshold" option as well?

File: src/sentry/relay/config.py (hunk 1 above)
Comment (jjbayer) [reply src/sentry/relay/config.py:413]: Keys are transaction names. So each transaction can have its own threshold defined.

File: src/sentry/relay/config.py (hunk 2 above)
Comment (shruthilayaj) [reply src/sentry/relay/config.py:444]: @jjbayer Yes, ProjectTransactionThreshold  and ProjectTransactionThresholdOverride are the only two places where transaction thresholds are defined/used from. For a particular transaction, if ProjectTransactionThreshold or ProjectTransactionThresholdOverride aren't specified, we fallback to a default value of 300ms applied on the transaction duration. I do still see some references to `sentry:apdex_threshold`  in the codebase,  I will make a PR today to remove those.

Comment (jjbayer) [reply src/sentry/relay/config.py:444]: Thank you!

**Review by untitaker — APPROVED** [review]

File: src/sentry/relay/config.py [hunk 3]
```diff
@@ -397,6 +400,50 @@ def _filter_option_to_config_setting(flt, setting):
...
+    # A separate default in Relay:
+    project_threshold: _TransactionThreshold = {
+        "metric": DEFAULT_THRESHOLD["metric"],
+        "threshold": float(DEFAULT_THRESHOLD["threshold"]),
+    }
+
+    # Apply custom project threshold
+    for i, threshold in enumerate(project.projecttransactionthreshold_set.all()):
+        if i > 0:
+            logger.error("More than one transaction threshold applied for project")
```
Comment (jjbayer) [inline src/sentry/relay/config.py:427]:
```suggestion
            logger.error("More than one transaction threshold exists for project")
```

**Pull request merged**